File: OtherLanguages/PHP/CrashReporter/Data/EcsResponse.php
<?
class EcsResponse
{
	private $status;
    private $data;
    private $message;

    public function __construct($status, $message, $data=null)
    {
        $this->status = $status;
        $this->message = $message;
        $this->data = $data;
    }
    
	public function write($writer)
	{
		$writer->writeValue(0, $this->status);
        $writer->writeString(1, $this->message);
        
        if ( !is_null($this->data) )
        {
            $writer->writeStructure(2, $this->data);
        }
	}
}
?>

File: Sitana.Framework/DataTransfer/Ecs/EcsResponse.cs
using System;

namespace Ebatianos.DataTransfer
{
	public class EcsResponse: IEcsStructure
	{
        public object StateObject;
        public int Status;

        public string Message;
        public IEcsStructure Data;

        void IEcsStructure.Read(EcsReader reader)
		{
			Status = reader.ReadInt32(0);
            Message = reader.ReadString(1);
            
            if (reader.HasField(2))
            {
                Data = reader.ReadStructure(2);
            }
		}

        void IEcsStructure.Write(EcsWriter writer)
		{
            throw new NotImplementedException("This structure is for read only.");
		}
	}
}